shared/x509utils/pem.go

```diff
@@ -106,3 +109,43 @@ func fileReadPEM(filename string, cb DecodePEMBlockFunc) (bool, error) {
 	// skip non-PEM files
 	return false, nil
 }
+
+// EncodeBytes produces a PEM encoded block
+func EncodeBytes(label string, body []byte, headers map[string]string) []byte {
+	var b bytes.Buffer
+	pem.Encode(&b, &pem.Block{
+		Type:    label,
+		Bytes:   body,
+		Headers: headers,
+	})
+	return b.Bytes()
+}
+
+// EncodePKCS1PrivateKey produces a PEM encoded RSA Private Key
+func EncodePKCS1PrivateKey(key *rsa.PrivateKey) []byte {
+	var out []byte
+	if key != nil {
+		body := x509.MarshalPKCS1PrivateKey(key)
+		out = EncodeBytes("RSA PRIVATE KEY", body, nil)
+	}
+	return out
+}
+
+// EncodePKCS8PrivateKey produces a PEM encoded Private Key
+func EncodePKCS8PrivateKey(key PrivateKey) []byte {
+	var out []byte
+	if key != nil {
+		body, err := x509.MarshalPKCS8PrivateKey(key)
+		if err != nil {
+			panic(err)
```

Comment:
Why calling panic if not recovering? If you do not want to recover than maybe Fatal(err) is a better option although I think that you should return a []byte and an error (or a bool ok)

Reply:
reason being function isn't expected to handle the case, and the panic being a form of "go and fix your f* code. Only good keys please. run .Validate() or something before calling me"

Reply:
in my book, you return errors when there is something to handle. this is a plain "F U"

Reply:
`x509.MarshalPKCS8PrivateKey` error value actually only happens due to key being a `any` and we are preventing that case by using our own PrivateKey interface here which is checked against the supported types on types.go

Reply:
works for me